Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp

```diff
@@ -359,125 +376,285 @@ template<RGBFunctionMode Mode> static Color parseRGBParametersRaw(CSSParserToken
-    // internal representation in engines.
-    return convertColor<SRGBA<uint8_t>>(HSLA<float> { static_cast<float>(normalizedHue), static_cast<float>(normalizedSaturation), static_cast<float>(normalizedLightness), static_cast<float>(alpha) });
+    return HSLA<float> {
+        static_cast<float>(resolvedHue),
+        static_cast<float>(resolvedSaturation),
+        static_cast<float>(resolvedLightness),
+        static_cast<float>(alpha)
+    };
 }
 
 static Color parseRelativeHSLParametersRaw(CSSParserTokenRange& args, ColorParserState& state)
 {
+    // <modern-hsl-syntax> = hsl([from <color>]?
+    //           [<hue> | none]
+    //           [<percentage> | <number> | none]
+    //           [<percentage> | <number> | none]
+    //           [ / [<alpha-value> | none] ]? )
+    // <modern-hsla-syntax> = hsla([from <color>]?
+    //         [<hue> | none]
+    //         [<percentage> | <number> | none]
+    //         [<percentage> | <number> | none]
+    //         [ / [<alpha-value> | none] ]? )
+
     ASSERT(args.peek().id() == CSSValueFrom);
     consumeIdentRaw(args);
 
     auto originColor = consumeOriginColorRaw(args, state);
     if (!originColor.isValid())
         return { };
 
-    auto originColorAsHSL = originColor.toColorTypeLossy<HSLA<float>>().resolved();
+    auto originColorAsHSL = originColor.toColorTypeLossy<HSLA<float>>();
+    auto originColorAsHSLResolved = originColorAsHSL.resolved();
 
     CSSCalcSymbolTable symbolTable {
-        { CSSValueH, CSSUnitType::CSS_DEG, originColorAsHSL.hue },
-        { CSSValueS, CSSUnitType::CSS_PERCENTAGE, originColorAsHSL.saturation },
-        { CSSValueL, CSSUnitType::CSS_PERCENTAGE, originColorAsHSL.lightness },
-        { CSSValueAlpha, CSSUnitType::CSS_PERCENTAGE, originColorAsHSL.alpha * 100.0 }
+        { CSSValueH, CSSUnitType::CSS_NUMBER, originColorAsHSLResolved.hue },
+        { CSSValueS, CSSUnitType::CSS_NUMBER, originColorAsHSLResolved.saturation },
+        { CSSValueL, CSSUnitType::CSS_NUMBER, originColorAsHSLResolved.lightness },
+        { CSSValueAlpha, CSSUnitType::CSS_NUMBER, originColorAsHSLResolved.alpha }
     };
 
     auto hue = consumeAngleOrNumberOrNoneRawAllowingSymbolTableIdent(args, symbolTable, state.mode);
     if (!hue)
         return { };
 
-    auto saturation = consumePercentOrNoneRawAllowingSymbolTableIdent(args, symbolTable);
+    auto saturation = consumeNumberOrPercentOrNoneRawAllowingSymbolTableIdent(args, symbolTable);
     if (!saturation)
         return { };
 
-    auto lightness = consumePercentOrNoneRawAllowingSymbolTableIdent(args, symbolTable);
+    auto lightness = consumeNumberOrPercentOrNoneRawAllowingSymbolTableIdent(args, symbolTable);
     if (!lightness)
         return { };
 
-    auto alpha = consumeOptionalAlphaRawAllowingSymbolTableIdent(args, symbolTable);
+    // NOTE: In the relative syntax, the alpha value of the origin color is used if no alpha value
```

Comment:
I find this comment a little unclear. Is it explaining \why the code below does this? Can you reword to make that more obvious?